exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h

```diff
@@ -0,0 +1,1077 @@
+    Properties *evt                   = dynamic_cast<Properties *>(&attribs);
+    if (evt != nullptr)
+    {
+      // Pass as a reference to original modifyable collection without creating a copy
+      return AddEvent(span, name, timestamp, *evt);
+    }
+#endif
+    // Pass a copy converted to Properties object on stack
+    Properties evtCopy = attributes;
+    return AddEvent(span, name, timestamp, evtCopy);
+  }
+
+  /**
+   * @brief Add event data to span associated with tracer.
+   * @param span Parent span.
+   * @param name Event name.
+   * @param timestamp Event timestamp.
+   * @param attributes Event attributes.
+   * @return
+   */
+  void AddEvent(trace::Span &span,
+                nostd::string_view name,
+                core::SystemTimestamp timestamp,
+                Properties &evt) noexcept
+  {
+    // TODO: respect originating timestamp. Do we need to reserve
+    // a special 'Timestamp' field or is it an overkill? The delta
+    // between when `AddEvent` API is called and when ETW layer
+    // timestamp is appended is nanos- to micros-, thus handling
+    // the explicitly provided timestamp is only necessary in case
+    // if a process wants to submit back-dated or future-dated
+    // timestamp. Unless there is a strong ask from any ETW customer
+    // to have it, this feature (custom timestamp) remains unimplemented.
+    (void)timestamp;
+
+    const auto &cfg = GetConfiguration(tracerProvider_);
+
+    evt[ETW_FIELD_NAME] = name.data();
+
+    const auto &spanContext = span.GetContext();
+    if (cfg.enableSpanId)
+    {
+      evt[ETW_FIELD_SPAN_ID] = ToLowerBase16(spanContext.span_id());
+    }
+
+    if (cfg.enableTraceId)
+    {
+      evt[ETW_FIELD_TRACE_ID] = ToLowerBase16(spanContext.trace_id());
+    }
+
+    LPGUID ActivityIdPtr = nullptr;
+    GUID ActivityId;
+    if (cfg.enableActivityId)
+    {
+      CopySpanIdToActivityId(spanContext, ActivityId);
+      ActivityIdPtr = &ActivityId;
+    }
+
+    etwProvider().write(provHandle, evt, ActivityIdPtr, nullptr, 0, encoding);
+  };
+
+  /**
+   * @brief Add event data to span associated with tracer.
+   * @param span Span.
+   * @param name Event name.
+   * @param timestamp Event timestamp.
+   * @return
+   */
+  void AddEvent(trace::Span &span,
+                nostd::string_view name,
+                core::SystemTimestamp timestamp) noexcept
+  {
+    AddEvent(span, name, timestamp, sdk::GetEmptyAttributes());
+  };
+
+  /**
+   * @brief Add event data to span associated with tracer.
+   * @param span Spab.
+   * @param name Event name.
+   */
+  void AddEvent(trace::Span &span, nostd::string_view name)
+  {
+    AddEvent(span, name, std::chrono::system_clock::now(), sdk::GetEmptyAttributes());
+  };
+
+  /**
+   * @brief Tracer destructor.
+   */
+  virtual ~Tracer() { CloseWithMicroseconds(0); };
+};
+
+/**
+ * @brief ETW::Span allows to send event data to ETW listener.
+ */
+class Span : public trace::Span
```

Comment:
I still don't fully understand why a `Span` and a `Recordable` is needed, and why not the `Recordable` passes data directly to the `Tracer`.

Reply:
Because `Recordable` is in SDK, not in API. And is not needed to implement this exporter. It is not used at all.